parse.go

```diff
@@ -64,9 +64,20 @@ type command struct {
 	dest        path
 	specs       []*spec
 	subcommands []*command
+	groups      []*command
```

Comment:
I have the sense that option groups should be represented by a new struct since they won't be allowed to contain subcommands, and won't need a name (I suppose)?

Reply:
Their behavior is too similar to either duplicate the logic and composing two types of structs would make the code too messy.

Instead I've refactored the struct parsing and tried to reduce complexity, while also adding a check that sub commands are not being nested under option groups.

Reply:
The name is used in the help message as the `{name} options:` header.